packages/api/src/pins.js

```diff
@@ -165,61 +162,46 @@ export async function pinPost (request, env, ctx) {
  * @return {Promise<JSONResponse>}
  */
 async function createPin (pinData, authToken, env, ctx) {
-  const { cid, name, origins, meta } = pinData
-  const normalizedCid = normalizeCid(cid)
-
-  const pinRequestData = {
-    requestedCid: cid,
-    cid: normalizedCid,
-    authKey: authToken
-  }
-  const pinOptions = {}
+  const { cid, origins, meta, normalizedCid } = pinData
 
-  if (name) {
-    pinRequestData.name = name
-    pinOptions.name = name
-  }
+  const pinName = pinData.name || undefined // deal with empty strings
 
-  if (origins) {
-    pinOptions.origins = origins
-  }
+  await env.cluster.pin(cid, {
+    name: pinName,
+    origins,
+    metadata: meta
```

Comment:
We can't pass user provided metadata to cluster. Cluster is not multi-tenant so two users pinning the same CID will ovewrite the meta.

Reply:
Really good catch, I'll update.